### PR TITLE
fix jenkins basic auth

### PR DESCRIPTION
**Pull request by shaowenchen**
Signed-off-by: shaowenchen <mail@chenshaowen.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

The apiserver uses Bearer Token , but Jenkins needs Basic Token. This pr transfers the Auth Token.

### CODE REVIEW

File: pkg/simple/client/devops/jenkins/pure_request.go [hunk 1]
```diff
@@ -24,10 +30,41 @@ func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *dev
 
 	client := &http.Client{Timeout: 30 * time.Second}
 
+	header := httpParameters.Header
+	bearTokenArray := strings.Split(header.Get("Authorization"), " ")
+	bearFlag := bearTokenArray[0]
```
Comment (zryfish) [inline pkg/simple/client/devops/jenkins/pure_request.go:35]: What if there is no `Authorization` header in the request? Better to check bearTokenArray first.

File: pkg/simple/client/devops/jenkins/pure_request.go [hunk 2]
```diff
@@ -24,10 +30,41 @@ func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *dev
...
+	if err == nil && bearFlag == "Bearer" {
```
Comment (zryfish) [inline pkg/simple/client/devops/jenkins/pure_request.go:36]: it's unnecessary to have `err == nil` since we returned if not.

Comment (zryfish) [reply pkg/simple/client/devops/jenkins/pure_request.go:36]: Also better to use case insensitive compare
```suggestion
	if strings.ToLower(bearFlag) == "bearer" {
```

File: pkg/simple/client/devops/jenkins/pure_request.go [hunk 3]
```diff
@@ -24,10 +30,41 @@ func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *dev
...
+		bearToken := bearTokenArray[1]
+		if err != nil {
+			klog.Error(err)
+			return nil, nil, err
+		}
+		infoArray := strings.Split(bearToken, ".")
```
Comment (zryfish) [inline pkg/simple/client/devops/jenkins/pure_request.go:42]: Why not using a tokenIssuer `pkg/apiserver/authentication/token/jwt.go` to parse a token?

File: pkg/simple/client/devops/jenkins/pure_request.go (hunk 1 above)
Comment (shaowenchen) [reply pkg/simple/client/devops/jenkins/pure_request.go:35]: Even if there is no Authorization header，bearTokenArray is assigned as [''] .

File: pkg/simple/client/devops/jenkins/pure_request.go (hunk 3 above)
Comment (shaowenchen) [reply pkg/simple/client/devops/jenkins/pure_request.go]: fixed

**Pull request merged**